src/dbnode/storage/index.go

```diff
@@ -1494,7 +1495,7 @@ func (i *nsIndex) query(
 		}
 
 		// NB(r): Make sure error is not retried and returns as bad request.
-		return exhaustive, xerrors.NewInvalidParamsError(fmt.Errorf(
+		return exhaustive, limits.NewQueryLimitExceededError(fmt.Sprintf(
```

Comment:
The error should be kept wrapped in `NewInvalidParamsError()`, since we use the it to differentiate between 4xx and 5xx errors. We do this in other places, too:

https://github.com/m3db/m3/blob/5480bcfea2dc586b868309bfb9ad5a018f6a3721/src/dbnode/storage/limits/query_limits.go#L295-L296

https://github.com/m3db/m3/blob/5480bcfea2dc586b868309bfb9ad5a018f6a3721/src/dbnode/storage/limits/query_limits.go#L306-L308